src/main/java/ca/corefacility/bioinformatics/irida/database/changesets/FastqcToFilesystem.java

```diff
@@ -0,0 +1,268 @@
+ * class (and out of the database).  This will instead save them to the filesystem as an {@link
+ * ca.corefacility.bioinformatics.irida.model.workflow.analysis.AnalysisOutputFile}.  This should greatly decrease the
+ * size of the database for large IRIDA installs and speed up loading of the {@link
+ * ca.corefacility.bioinformatics.irida.model.workflow.analysis.AnalysisFastQC} class.
+ */
+public class FastqcToFilesystem implements CustomSqlChange {
+
+	//going to do the updates in batches so things go smoothly
+	private final long BATCH_SIZE = 10000;
+
+	private static final Logger logger = LoggerFactory.getLogger(FastqcToFilesystem.class);
+	private Path outputFileDirectory;
+	private Path tempOutputDirectory;
+
+	private DataSource dataSource;
+
+	@Override
+	public SqlStatement[] generateStatements(Database database) throws CustomChangeException {
+
+		JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+
+		Long previousMaxId = getMaxId(jdbcTemplate);
+
+		List<Path> writtenFiles = new ArrayList<>();
+
+		//create a temp directory for fastqc files
+		try {
+			this.tempOutputDirectory = Files.createTempDirectory(outputFileDirectory, "irida-fastqc-temp");
+		} catch (IOException e) {
+			throw new CustomChangeException("Could not create temp directory for fastqc files", e);
+		}
+
+		logger.info("This update will be writing files temporarily to " + tempOutputDirectory
+				+ ".  They should be automatically transferred to your analysis output directory " + outputFileDirectory
+				+ "  upon completion.");
+
+		//wrapping everything in a try/catch so we can log a nice error message in case of a problem.  Err
+		try {
+
+			//the chart types we're taking out of the database
+			List<String> chartTypes = Lists.newArrayList("perBaseQualityScoreChart", "perSequenceQualityScoreChart",
+					"duplicationLevelChart");
+
+			//inserting empty output files for charts
+			chartTypes.forEach(chart -> {
+				int update = jdbcTemplate.update(
+						"INSERT INTO analysis_output_file (created_date, execution_manager_file_id, analysis_id) SELECT createdDate, '"
+								+ chart + ".png', a.id FROM analysis a INNER JOIN analysis_fastqc q ON a.id=q.id");
+
+				logger.info("Inserted " + update + " temp output file entries for chart type " + chart);
+			});
+
+			//for each chart type, write the new files
+			chartTypes.forEach(chart -> {
+				//write the files and update the analysis_output_file directory
+				writtenFiles.addAll(writeFileForChartType(chart, jdbcTemplate));
+
+				//Add entries to analysis_output_file_map linking to the new analysis_output_file entries
+				jdbcTemplate.update(
+						"insert into analysis_output_file_map (analysis_id, analysisOutputFilesMap_id, analysis_output_file_key) SELECT analysis_id, id, '"
+								+ chart + "' FROM analysis_output_file where execution_manager_file_id='" + chart
+								+ ".png'");
+			});
+
+		} catch (RuntimeException e) {
```

Comment:
Why only `RuntimeException` instead of `Exception`, since this will miss some exceptions. In particular `IOException` and `CustomChangeException`s are not `RuntimeException`.

Reply:
No problem to bump up to Exception.  Done in 225a45d7e1